automation/utils/slack.py

```diff
@@ -0,0 +1,114 @@
+import enum
+from typing import Optional
+
+import pydantic
+import slack_sdk
+
+from .. import config
+
+##########
+# MODELS #
+##########
+
+
+class Channel(pydantic.BaseModel):
+    created: int
+    creator: str
+    id: str
+    is_archived: bool
+    is_channel: bool
+    is_ext_shared: bool
+    is_general: bool
+    is_group: bool
+    is_im: bool
+    is_member: bool
+    is_mpim: bool
+    is_org_shared: bool
+    is_pending_ext_shared: bool
+    is_private: bool
+    is_shared: bool
+    name: str
+    name_normalized: str
+    num_members: int
+
+
+class Message(pydantic.BaseModel):
+    text: str
+    ts: float
+    user: Optional[str]
+    type: str
+    subtype: Optional[str]
+
+
+class Profile(pydantic.BaseModel):
+    display_name: Optional[str]
+    email: str
+
+
+class User(pydantic.BaseModel):
+    id: str
+    is_admin: bool = False
+    is_bot: bool = False
+    is_owner: bool = False
+    name: str
+    real_name: Optional[str]
+    deleted: bool = False
+    profile: Profile
+
+    def get_handle(self):
+        # TODO : this logic was copied from the vintage automation, reverify it
+        return "@" + (
+            self.profile.display_name
+            if self.profile.display_name is not None
+            else self.real_name
+        )
+
+
+##########
+# CLIENT #
+##########
+
+_USERS_NOT_FOUND = "users_not_found"
+
+
+class SlackErrors(enum.Enum):
+    USERS_NOT_FOUND = "users_not_found"
+
+
+# TODO : add support for other slack client methods
+class SlackClient:
+    def __init__(self):
+        self._slack_sdk_client = slack_sdk.WebClient(
+            token=config.Config.load().slack.api_key
+        )
+
+    def _slack_sdk_wrapper(
```

Comment:
I guess maybe over time we'll learn which parts of the slack API we're going to use and can make our wrapper more powerful. I think it would be cool to eventually have an interface like this:

```
slack = SlackClient()
user = slack.users.find(email='leif@example.com')
```

Reply:
I agree that its unnecessarily complex rn. I have a messy uncommited slack client that has more functionality, including logic to handle the slack API's pagination and rate limiting, which is why I went with the wrapper function. Let's revisit this again once the client is more fleshed out.

Reply:
I agree that its unnecessarily complex rn. I have a messy uncommited slack client that has more functionality, including logic to handle the slack API's pagination and rate limiting, which is why I went with the wrapper function.

I like that interface. I'll add a comment suggesting to remind us to go with that approach when we flesh out the client.